storage/innobase/dict/dict0dd.cc

```diff
@@ -981,7 +982,11 @@ dict_table_t *dd_table_create_on_dd_obj(const dd::Table *dd_table,
         if (c->is_virtual() == dd_col->is_virtual()) col_pos++;
       }
 
-      bool is_asc = (idx_elem->order() == dd::Index_element::ORDER_ASC);
+      /* FULLTEXT and HASH indexes can have UNDEF order, we should treat UNDEF
```

Comment:
Just realized that you mentioned HASH indexes here , can you please cover this in the testcase?
Also add a spatial index as well https://dev.mysql.com/doc/refman/8.0/en/creating-spatial-indexes.html